bluesky/plan_patterns.py

```diff
@@ -67,6 +67,99 @@ def spiral(x_motor, y_motor, x_start, y_start, x_range, y_range, dr, nth, *,
     return cyc
 
 
+def spiral_square_pattern(x_motor, y_motor, x_center, y_center, x_range, y_range, x_num,y_num):
+    '''Square spiral scan, centered around (x_start, y_start)
+
+    Parameters
+    ----------
+    x_motor : object, optional
+        any 'setable' object (motor, temp controller, etc.)
+    y_motor : object, optional
+        any 'setable' object (motor, temp controller, etc.)
+    x_center : float
+        x center
+    y_center : float
+        y center
+    x_range : float
+        x width of spiral
+    y_range : float
+        y width of spiral
+    x_num : float
+        number of x axis points
+    y_num : float (must be even if x_num is even and must be odd if x_num is odd, if not it is increased by 1 to ensure this)
+        number of y axis points
+
+    Returns
+    -------
+    cyc : cycler
+    '''
+    x_points, y_points = [], []
+
+    if x_num%2==0:
+        num_st= 2
+        if y_num%2==1:
+           y_num+=1
+        offset=0.5
+    else:
+        num_st=1
+        if y_num%2==0:
+            y_num+=1
+        offset=0
+
+        x_points.append(x_center)
+        y_points.append(y_center)
+
+    delta_x = x_range/(x_num-1)
+    delta_y = y_range/(y_num-1)
+    num_ring = max(x_num,y_num)
+    x_max=x_center + delta_x * (x_num-1)/2
+    x_min=x_center - delta_x * (x_num-1)/2
+    y_max=y_center + delta_y * (y_num-1)/2
+    y_min=y_center - delta_y * (y_num-1)/2
+
+    for n,i_ring in enumerate(range(num_st, num_ring+1,2)):
+        x_ring_max=x_center + delta_x * (n+offset)
+        y_ring_max=y_center + delta_y * (n+offset)
+        x_ring_min=x_center - delta_x * (n+offset)
+        y_ring_min=y_center - delta_y * (n+offset)
+
+        for n in range(1, i_ring):
+            x = x_ring_min+delta_x*(n-1)
+            y = y_ring_min
+
+            if ( (x <= x_max) and (x>= x_min) and (y<=y_max) and (y>=y_min)     ):
```

Comment:
The outer parens here aren't technically necessary. If you want them for readability's sake, I'd say that's a judgment call, but PEP8 would have you remove the extra  spaces inside the outer parens.

Reply:
I'd recommend using [flake8](https://pypi.python.org/pypi/flake8), which hits all inconsistencies with PEP8. We will meet with @awalter-bnl later today to discuss/fix it.